scheduler/controllers/odigosconfig/odigosconfig_controller.go

```diff
@@ -41,12 +42,31 @@ func (r *odigosConfigController) Reconcile(ctx context.Context, _ ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	odigosDeployment := corev1.ConfigMap{}
+	err = r.Client.Get(ctx, types.NamespacedName{Namespace: env.GetCurrentNamespace(), Name: k8sconsts.OdigosDeploymentConfigMapName}, &odigosDeployment)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	// effective profiles are what is actually used in the cluster (minus non existing profiles and plus dependencies)
 	availableProfiles := profiles.GetAvailableProfilesForTier(r.Tier)
-	effectiveProfiles := calculateEffectiveProfiles(odigosConfig.Profiles, availableProfiles)
+
+	allProfiles := make([]common.ProfileName, 0)
+	allProfiles = append(allProfiles, odigosConfig.Profiles...)
+
+	tokenProfilesString := odigosDeployment.Data[k8sconsts.OdigosDeploymentConfigMapOnPremClientProfilesKey]
```

Comment:
it's considered more idomatic to use the second value `ok` when accessing a map key, instead of comparing the result to empty string